include/envoy/server/filter_config.h

```diff
@@ -90,11 +90,12 @@ class ProtocolOptionsFactory : public Config::TypedFactory {
    * @param validation_visitor message validation visitor instance.
    * @return Upstream::ProtocolOptionsConfigConstSharedPtr the protocol options
    */
-  virtual Upstream::ProtocolOptionsConfigConstSharedPtr
-  createProtocolOptionsConfig(const Protobuf::Message& config,
-                              ProtobufMessage::ValidationVisitor& validation_visitor) {
+  virtual Upstream::ProtocolOptionsConfigConstSharedPtr createProtocolOptionsConfig(
+      const Protobuf::Message& config,
+      Server::Configuration::TransportSocketFactoryContext& factory_context) {
```

Comment:
Can you define a new type like `ProtocolOptionsFactoryContext`? It can be just type aliased now, i.e. `using ProtocolOptionsFactoryContext = TransportSocketFactoryContext;` in `include/envoy/server/factory_context.h`